iib/workers/tasks/opm_operations.py

```diff
@@ -295,196 +295,132 @@ def inner(*args, **kwargs):
-        '-t',
-        '/dev/null',
-    ]
-
-    if opm_pprof_port:
-        # by default opm uses the 127.0.0.1:6060
-        cmd.extend(["--pprof-addr", f"127.0.0.1:{str(opm_pprof_port)}"])
-
-    cwd = os.path.abspath(os.path.join(catalog_dir, os.path.pardir))
-    result = (
-        opm_port,
-        _serve_cmd_at_port_defaults(cmd, cwd, opm_port),
-    )
-    return result
+    error_msg = "No olm package version found for OLM bundle."
+    log.warning(error_msg)
+    raise IIBError(error_msg)
 
 
-@create_port_filelocks(port_purposes=["opm_port"])
-def opm_registry_serve(
-    opm_port: int,
-    db_path: str,
-) -> Tuple[int, subprocess.Popen]:
+def _get_input_data_path(input_data: str, base_dir: str) -> str:
     """
-    Locally start OPM registry service, which can be communicated with using gRPC queries.
+    Retrieve correct path to date which will be used when calling opm render.
 
-    Due to IIB's paralellism, the service can run multiple times, which could lead to port
-    binding conflicts. Resolution of port conflicts is handled in this function as well.
+    If input_data is FBC index image we extract configs folder and return path to it.
+    If input_data is not FBC we extract hidden database and return path to it.
 
-    :param int opm_port: OPM port number obtained from create_port_filelock decorator
-    :param int opm_pprof_port: Pprof opm port number obtained from create_port_filelock decorator
-    :param str db_path: path to index database containing the registry data.
-    :return: tuple containing port number of the running service and the running Popen object.
-    :rtype: (int, Popen)
+    :param str input_data: input data
+    :param str base_dir: temp directory where data are located.
+
+    :return: path to input for opm render
+    :rtype: str
     """
-    log.info('Serving data from index.db %s', db_path)
+    if os.path.exists(input_data):
+        return input_data
 
-    cmd = [
-        Opm.opm_version,
-        'registry',
-        'serve',
-        '-p',
-        str(opm_port),
-        '-d',
-        db_path,
-        '-t',
-        '/dev/null',
-    ]
+    if not is_image_fbc(input_data):
+        from iib.workers.tasks.build import _get_index_database
 
-    cwd = os.path.dirname(db_path)
-    result = (
-        opm_port,
-        _serve_cmd_at_port_defaults(cmd, cwd, opm_port),
-    )
-    return result
+        log.info('Extracting SQLite DB from image %s', input_data)
+        return _get_index_database(input_data, base_dir)
+    else:
+        log.info('Extracting FBC from image %s', input_data)
+        return get_catalog_dir(input_data, base_dir)
 
 
-def _serve_cmd_at_port_defaults(serve_cmd: List[str], cwd: str, port: int) -> subprocess.Popen:
+def get_list_bundles(
+    input_data: str,
+    base_dir: str,
+) -> List[BundleImage]:
     """
-    Call `_serve_cmd_at_port()` with default values from IIB config.
+    Run OPM render, which can be communicated with using gRPC queries.
```

Comment:
can you please explan how gRPC queries is relevant here? are we still using gPRC anywhere?

Reply:
Thank you for noticing.